sbnana/CAFAna/Core/EnsembleRatio.cxx

```diff
@@ -142,6 +142,39 @@ namespace ana
     return g;
   }
 
+  //----------------------------------------------------------------------
+  Eigen::MatrixXd EnsembleRatio::CovarianceMatrix()
+  {
+    // TODO there is probably a better way to check this?
+    assert (fMultiverse->ShortName().substr(0,3) == "gas");
```

Comment:
This is too ugly to live. Can you add an enum to FitMultiverse that records how it was initialized, and a function to access it?
I had something like that in an early draft, and then decided it was unnecessary. I think this is a reasonable justification for it. The other option could be to iterate through the universes and check that none of the shifts is zero (should be vanishing probability if they are thrown randomly).